resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -793,6 +793,25 @@ private[spark] class Client(
       // distributed file.
       amKeytabFileName.foreach { kt => props.setProperty(KEYTAB.key, kt) }
 
+      // Upload user provided ivysettings.xml file to the distributed cache
+      val ivySettings = sparkConf.getOption("spark.jars.ivySettings")
+      if (isClusterMode && ivySettings.isDefined) {
+        val ivySettingsFile = new File(ivySettings.get)
+        require(ivySettingsFile.exists(), s"Ivy settings file $ivySettingsFile not found")
+        require(ivySettingsFile.isFile(),
+          s"Ivy settings file $ivySettingsFile is not a normal file")
+        // Generate a file name that can be used for the ivySettings file, that does not conflict
+        // with any other conf file.
+        val amIvySettingsFileName = ivySettingsFile.getName() + "-" + UUID.randomUUID().toString
+        confStream.putNextEntry(new ZipEntry(amIvySettingsFileName))
+        Files.copy(ivySettingsFile, confStream)
```

Comment:
Can we use `java.nio.file.Files` here instead of Guava's `Files`? I know Guava is already used in this file but I think it's better to leverage the built-in functionality moving forward. You can do a rename import like `import java.nio.file.{Files => NioFiles}` (or update the other references as well if it's a small change?)

Reply:
Done